clients/src/main/java/org/apache/kafka/common/errors/OffsetOutOfRangeException.java

```diff
@@ -35,4 +43,8 @@ public OffsetOutOfRangeException(String message, Throwable cause) {
         super(message, cause);
     }
 
+    public Map<TopicPartition, Long> getOutOfRangePartitions() {
```

Comment:
@ijuma Thanks much for your review. Will pay attention to final, private and get method name in the future patch.

Please see updated patch. I didn't add final to variable offsetOutOfRangePartitions in class OffsetOutOfRangeException because otherwise I need to add 4 additional lines offsetOutOfRangePartitions=null to 4 constructor method. This seems redundant. Can you let me know if you prefer me adding final keyword and the additional 4 lines?